src/player.cpp

```diff
@@ -3618,15 +3620,51 @@ void player::apply_damage(Creature *source, body_part hurt, int dam)
         hp_cur[hurtpart] = 0;
     }
 
-    if( hp_cur[hurtpart] <= 0 && ( source == nullptr || !source->is_hallucination() )) {
-        remove_effect( effect_mending, hurt );
-        add_effect( effect_disabled, 1_turns, hurt, true );
+    if( hp_cur[hurtpart] <= 0 && ( source == nullptr || !source->is_hallucination() ) ) {
+        if( has_effect( effect_mending, hurt ) ) {
+            effect &e = get_effect( effect_mending, hurt );
+            float remove_mend = dam / 20.0f;
+            e.mod_duration( -e.get_max_duration() * remove_mend );
+        }
     }
 
     lifetime_stats.damage_taken += dam;
     if( dam > get_painkiller() ) {
         on_hurt( source );
     }
+
+    // remove healing effects if damaged
+    int remove_med = roll_remainder( dam / 5.0f );
+    int bandaged_intensity = 0;
+    int disinfected_intensity = 0;
+
+    if( remove_med > 0 && has_effect( effect_bandaged, hurt ) ) {
+        effect &e_bandaged = get_effect( effect_bandaged, hurt );
+        bandaged_intensity = e_bandaged.get_intensity();
+        if( remove_med >= bandaged_intensity ) {
+            remove_effect( effect_bandaged, hurt );
+            remove_med -= bandaged_intensity;
+            add_msg_if_player( m_bad, _( "Bandages on your %s was destroyed!" ), body_part_name( hurt ) );
```

Comment:
Is it plural or singular?
Bandages were? Or (one) bandage was?

Reply:
Dont ask me about Boston accent. Me not understand. Just say the correct version and I will change it, because both variant sounds good for me.

Reply:
Decide on one of "(one) bandage was destroyed" or "(multiple) bandages were destroyed".

Reply:
I would go with "Bandages on your %s were destroyed!"